drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -123,7 +170,8 @@ static void adc_context_start_sampling(struct adc_context *ctx)
 		val = GPADC->GP_ADC_CTRL_REG & ~PER_CHANNEL_ADC_CONFIG_MASK;
 		val |= m_channels[current_channel].gp_adc_ctrl_reg;
 		val |= GPADC_GP_ADC_CTRL_REG_GP_ADC_START_Msk |
-		       GPADC_GP_ADC_CTRL_REG_GP_ADC_MINT_Msk;
+		       GPADC_GP_ADC_CTRL_REG_GP_ADC_MINT_Msk |
+			   GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
```

Comment:
`ADC_LDO_1V2 Start-Up Delay` is 22 us. We should wait before sampling starts.
Should we move it back to  `gpadc_smartbond_resume()`?

Reply:
Nice catch! We should also take into account the fact that `ADC_LDO_1V2` consumes approx. 5uA. Following is an excerpt from the DS revision 3.3 (section 27.2.2.1):
```
The GPADC is a dynamic ADC and consumes no static power, except for the ADC_LDO_1V2 that
consumes approximately 5 μA. Therefore, GP_ADC_EN must be set to 0 if the ADC is not used.
```
Therefore, powering on GPADC even if no conversations are requested, there should be a penalty in power consumption. On the other hand, and with the current implementation, we should add the worst case delay before sampling (I do not see any status register that reflectS the readiness of the mentioned LDO).

Reply:
Moved to gpadc_smartbond_resume, that gets called in start_read so wasted power should be negligible. But enough time for it to settle.

Reply:
@NiekIlmer your statement above should be valid in device runtime mode. In non-runtime mode, PM should automatically resume ADC upon exiting the sleep state. Inevitably, there should be a trade off between latency and power consumption. Personally, I am OK with the current approach.